(GH-102) Prototype streaming remote output (#2608)
puppetlabs/bolt: documentation/experimental_features.md

Bolt team attempts to make these changes painless by providing useful warnings
around breaking behavior where possible. 

## Streaming output

This feature was introduced in [Bolt 3.2.0](https://github.com/puppetlabs/bolt/blob/main/CHANGELOG.md#bolt-310-2021-3-08).

The new `stream` output option can be set in [transport

> configuration](bolt_transports_reference.md). Bolt will stream results back to the console as they

Reviewer: It would be nice to have this is a CLI option as well, so it can be toggled easily. Should be as simple as adding a switch to the option parser since the logic for translating it to transport config is already there.

Author: Yeah - I also realized that now that it's added to all the transports (it was originally just Bash shell) it can also be a Bolt config option, instead of transport config. Refactored to suit.

Reviewer: Ahh, that feels _much_ better than having it be a transport option. :) Good idea.

> The new `stream` output option can be set in either `bolt-project.yaml` or `bolt-defaults.yaml`, or
> specified on the command line as `--stream`. Bolt will stream results back to the console as they
> are received, with the target's safe name (the URI without the password included) and the stream
> (either 'out' or 'err') appended to the message, like so:

Reviewer 2: ```suggestion
You can set the new `stream` output option in `bolt-project.yaml` or `bolt-defaults.yaml`, or
specify the option on the command line as `--stream`. Bolt streams results back to the console as they
are received, with the target's safe name (the URI without the password included) and the stream
(either 'out' or 'err') appended to the message, like so:
```

```
Started on docker://puppet_6_node...
Started on docker://puppet_7_node...
[docker://puppet_7_node] out: Hello!
[docker://puppet_6_node] out: Hello!
Finished on docker://puppet_7_node:
  Hello!
Finished on docker://puppet_6_node:
  Hello!
```

> As you can see above, when streaming is configured output may be printed to the console twice: as
> the actions are running, and again when results are printed.

Reviewer 2: ```suggestion
As you can see, when you configure output to stream, Bolt might print to the console twice:
once as the actions are running, and again after Bolt prints the results.
```
